tabular/src/autogluon/tabular/predictor/predictor.py

```diff
@@ -954,6 +1022,181 @@ def fit_extra(self, hyperparameters, time_limit=None, base_model_names=None, **k
+                logger.log(20,
+                           f'Pseudolabeling algorithm confidently assigned pseudolabels to: {len(test_pseudo_idxes_true)} rows of data'
+                           f'on iteration: {iter_print}. Adding to train data')
+
+            test_pseudo_idxes = pd.Series(data=False, index=y_pred_proba.index)
+            test_pseudo_idxes[test_pseudo_idxes_true.index] = True
+
+            y_pseudo_og = y_pseudo_og.append(y_pred.loc[test_pseudo_idxes_true.index], verify_integrity=True)
+
+            pseudo_data = unlabeled_data.loc[y_pseudo_og.index]
+            pseudo_data[self.label] = y_pseudo_og
+            self.fit_extra(pseudo_data=pseudo_data, name_suffix=PSEUDO_MODEL_SUFFIX.format(iter=(i + 1)),
+                           **kwargs)
+            current_score = self.info()['best_model_score_val']
+
+            logger.log(20,
+                       f'Pseudolabeling algorithm changed validation score from: {previous_score}, to: {current_score}'
+                       f' using evaluation metric: {self.eval_metric.name}')
+
+            if previous_score >= current_score:
+                break
+            else:
+                # Cut down X_test to not include pseudo labeled data
+                X_test = X_test.loc[test_pseudo_idxes[~test_pseudo_idxes].index]
+                previous_score = current_score
+
+        if return_pred_prob:
+            return self, y_pred_proba_og
+        else:
+            return self
+
+    def fit_pseudolabel(self, pseudo_data: pd.DataFrame, max_iter: int = 5, return_pred_prob: bool = False,
+                        use_ensemble: bool = False, **kwargs):
+        """
+        If 'pseudo_data' is labeled then incorporates all test_data into train_data for
+        newly fit models. If 'pseudo_data' is unlabeled then 'fit_pseudolabel' will self label the
+        data and will augment the original training data by adding all the self labeled
+        data that meets a criteria (For example all rows with predictive prob above 95%). If
+        predictor is fit then will call fit_extra with added training data, if predictor
+        is not fit then will fit model on train_data then run.
+
+        Parameters
+        ----------
+        pseudo_data : str or :class:`TabularDataset` or :class:`pd.DataFrame`
+            Extra data to incorporate into training. Pre-labeled test data allowed. If no labels
+            then pseudolabeling algorithm will predict and filter out which rows to incorporate into
+            training
+        max_iter: int, default = 5
+            Maximum iterations of pseudolabeling allowed
+        return_pred_prob: bool, default = False
+            Returns held-out predictive probabilities from pseudo-labeling. If test_data is labeled then
+            returns model's predictive probabilities.
+        use_ensemble: bool, defautl = False
+            Flag to determine whether to use ensemble pseudo labeling algorithm
+        kwargs: dict
```

Comment:
clarify these are kwargs for either `fit()` or `fit_extra()`.

Reply:
"        kwargs: dict
            If predictor is not already fit, then kwargs are for the functions 'fit' and 'fit_extra':
            Refer to parameters documentation in :meth:`TabularPredictor.fit`.
            Refer to parameters documentation in :meth:`TabularPredictor.fit_extra`.
            If predictor is fit kwargs are for 'fit_extra':
            Refer to parameters documentation in :meth:`TabularPredictor.fit_extra`."